### PR TITLE
Add GitHub action for macOS builds

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

**Comment by stweil** [comment]
This pull request depends on pull request #410 and currently ignored errors from `make check` because of issue https://github.com/OCR-D/ocrd_froc/issues/10.

It must be triggered manually, but could also be run for example weekly in the future to detect any regressions earlier.

**Comment by stweil** [comment]
I tried two different variants for getting the submodules:

1. action/checkout can get all submodules recursively. By default it uses depth 1, but as it gets more submodules than required, it takes 189 s for the checkout step.
2. Get submodules during the build process (`make all GIT_DEPTH="--depth 1"`). That costs about 45 s during  the build step, but reduces the time for the checkout step to 1 s.

So the. 2nd variant is much better, at least with `GIT_DEPTH="--depth 1"` which should be used by all CI and Docker builds. Currently those builds don't use it.

**Comment by stweil** [comment]
Example result for this action: https://github.com/stweil/ocrd_all/actions/runs/8294000625/job/22698144574.

File: .github/workflows/makeall-macos.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      run: ls -l /usr/bin/python* /usr/local/bin/p*
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      # curl and icu4c are already installed
+      run: brew install automake bash cabextract geos leptonica libarchive libtool pango parallel
+    - name: Make all
+      if: runner.os == 'macOS'
+      run: LC_ALL=C PKG_CONFIG_PATH=$HOMEBREW_PREFIX/opt/icu4c/lib/pkgconfig:$HOMEBREW_PREFIX/opt/libarchive/lib/pkgconfig make all GIT_DEPTH="--depth 1"
+    - name: Make check
+      run: make check || echo "make check failed"
```
Comment (bertsky) [inline .github/workflows/makeall-macos.yml:78]:
```suggestion
      run: make check
```

File: .github/workflows/makeall-macos.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          - 'macos-13'
+          - 'macos-14'
+      python-version:
+        description: 'Python version'
+        required: true
+        default: 'default'
+        type: choice
+        options:
+          - 'default'
+          - '3.7'
```
Comment (bertsky) [inline .github/workflows/makeall-macos.yml:36]:
```suggestion
```

File: .github/workflows/makeall-macos.yml (hunk 1 above)
Comment (stweil) [reply .github/workflows/makeall-macos.yml]: Done. Note that the check will now fail until ocrd_froc was updated.

File: .github/workflows/makeall-macos.yml (hunk 2 above)
Comment (stweil) [reply .github/workflows/makeall-macos.yml]: Done, thank you for your review.

**Pull request merged**